src/phpDocumentor/Transformer/Template/TemplateNotFound.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link http://phpdoc.org
+ */
+
+namespace phpDocumentor\Transformer\Template;
+
+use InvalidArgumentException;
+use function sprintf;
+
+/**
+ * @codeCoverageIgnore not worth it
+ */
+final class TemplateNotFound extends InvalidArgumentException
```

Comment:
Would it make sense to have a look at our internal exceptions at some point? I get a feeling that we might want to add more customized exceptions to get more usefull feedback to our users with less work. 
By adding a `TemplateException` interface for example to this exception we can catch that interface to provide feedback about where things when wrong. 

This is not something that we should start doing now. But it would be an improvement on the overall code.

Reply:
I think it would; AFAIC we could best start with specific exceptions and only introduce base exceptions where needed. This, because I have bad experiences with exceptions based on a module (e.g. TemplateException) because it usually makes more sense to extend from InvalidArgumentException, RuntimeException, etc.

Reply:
For modules, it may be more interesting to use interfaces instead of superclasses